src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecutionTest.java

```diff
@@ -348,6 +354,25 @@ private static void assertStepExecutions(FlowExecution e, String... steps) throw
         });
     }
 
+    @Issue("JENKINS-46961")
+    @Test public void interruptedWhileStarting() throws Exception {
+        story.then(r -> {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
```

Comment:
TODO: Check if `PERFORMANCE_OPTIMIZED` affects the behavior:

```suggestion
            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
            p.addProperty(new DurabilityHintJobProperty(FlowDurabilityHint.PERFORMANCE_OPTIMIZED));
```

Reply:
It does not significantly change the behavior, but it does change the specific exception that is thrown. We should test both cases. Here is the exception with `MAX_SURVIVABILITY`:

```
java.nio.channels.ClosedByInterruptException
    at java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:202)
    at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:216)
    at hudson.util.FileChannelWriter.write(FileChannelWriter.java:72)
    at java.io.Writer.write(Writer.java:192)
    at hudson.util.AtomicFileWriter.write(AtomicFileWriter.java:162)
    at java.io.Writer.write(Writer.java:157)
    at hudson.XmlFile.write(XmlFile.java:189)
    at org.jenkinsci.plugins.workflow.support.PipelineIOUtils.writeByXStream(PipelineIOUtils.java:30)
    at org.jenkinsci.plugins.workflow.support.storage.SimpleXStreamFlowNodeStorage.writeNode(SimpleXStreamFlowNodeStorage.java:182)
    at org.jenkinsci.plugins.workflow.support.storage.SimpleXStreamFlowNodeStorage.flushNode(SimpleXStreamFlowNodeStorage.java:151)
    at org.jenkinsci.plugins.workflow.support.storage.SimpleXStreamFlowNodeStorage.storeNode(SimpleXStreamFlowNodeStorage.java:125)
    at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution$TimingFlowNodeStorage.storeNode(CpsFlowExecution.java:1827)
    at org.jenkinsci.plugins.workflow.cps.FlowHead.newStartNode(FlowHead.java:116)
    at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:530)
    at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:337)
    at hudson.model.ResourceController.execute(ResourceController.java:97)
    at hudson.model.Executor.run(Executor.java:429)
```

Here is the exception with `PERFORMANCE_OPTIMIZED`:

```
java.nio.channels.ClosedChannelException
    at sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:110)
    at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:199)
    at java.nio.channels.Channels.writeFullyImpl(Channels.java:78)
    at java.nio.channels.Channels.writeFully(Channels.java:101)
    at java.nio.channels.Channels.access$000(Channels.java:61)
    at java.nio.channels.Channels$1.write(Channels.java:174)
    at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
    at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
    at sun.nio.cs.StreamEncoder.implFlush(StreamEncoder.java:297)
    at sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:141)
    at java.io.OutputStreamWriter.flush(OutputStreamWriter.java:229)
    at com.thoughtworks.xstream.core.util.QuickWriter.flush(QuickWriter.java:75)
Also:   java.nio.channels.ClosedChannelException
        at sun.nio.ch.FileChannelImpl.ensureOpen(FileChannelImpl.java:110)
        at sun.nio.ch.FileChannelImpl.write(FileChannelImpl.java:199)
        at java.nio.channels.Channels.writeFullyImpl(Channels.java:78)
        at java.nio.channels.Channels.writeFully(Channels.java:101)
        at java.nio.channels.Channels.access$000(Channels.java:61)
        at java.nio.channels.Channels$1.write(Channels.java:174)
        at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
        at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
        at java.io.FilterOutputStream.close(FilterOutputStream.java:158)
        at org.jenkinsci.plugins.workflow.support.PipelineIOUtils.writeByXStream(PipelineIOUtils.java:35)
Caused: com.thoughtworks.xstream.io.StreamException:  : null
    at com.thoughtworks.xstream.core.util.QuickWriter.flush(QuickWriter.java:77)
    at com.thoughtworks.xstream.io.xml.PrettyPrintWriter.flush(PrettyPrintWriter.java:346)
    at com.thoughtworks.xstream.XStream.toXML(XStream.java:990)
    at hudson.util.XStream2.toXMLUTF8(XStream2.java:313)
    at org.jenkinsci.plugins.workflow.support.PipelineIOUtils.writeByXStream(PipelineIOUtils.java:34)
    at org.jenkinsci.plugins.workflow.support.storage.BulkFlowNodeStorage.flush(BulkFlowNodeStorage.java:168)
    at org.jenkinsci.plugins.workflow.support.storage.BulkFlowNodeStorage.storeNode(BulkFlowNodeStorage.java:142)
    at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution$TimingFlowNodeStorage.storeNode(CpsFlowExecution.java:1827)
    at org.jenkinsci.plugins.workflow.cps.FlowHead.newStartNode(FlowHead.java:116)
    at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:530)
    at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:337)
    at hudson.model.ResourceController.execute(ResourceController.java:97)
    at hudson.model.Executor.run(Executor.java:429)
```

Reply:
Hmm, actually it seems like there is an important difference between the durability levels. When using `MAX_SURVIVABILITY`, clearing the interruption status of the thread via `Thead.interrupted` when an exception is thrown in `CpsFlowExecution.start` is enough to fix the issue, but with `PERFORMANCE_OPTIMIZED`, the problem still occurs even with that change.